### PR TITLE
[TestOnly] Add multithreaded iOS tests

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Add multithreaded iOS tests, mainly for Trezor validation; BTC address derivation, BNB signing.

## Testing instructions

ios-test

## Types of changes

* New tests

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: swift/Tests/Addresses/BitcoinAddressTests.swift [hunk 1]
```diff
@@ -228,4 +228,41 @@ class BitcoinAddressTests: XCTestCase {
...
+    }
+
+    @objc func testDeriveOneThread() {
+        let n = 200
+        for _ in 1...n {
+            testBitcoinDeriveAddress()
+        }
+    }
+
+    func testMultiThreadedDerive() {
```
Comment (hewigovens) [inline swift/Tests/Addresses/BitcoinAddressTests.swift:245]: there are a few other ways (`DispatchGroup`, `DispatchSemaphore` etc) to write elegant concurrent code in swift

Comment (catenocrypt) [reply swift/Tests/Addresses/BitcoinAddressTests.swift:245]: will check

Comment (catenocrypt) [reply swift/Tests/Addresses/BitcoinAddressTests.swift:245]: Changed tests to use the more elegant `OperationQueue`

**Review by vikmeup — APPROVED** [review]

**Review by hewigovens — APPROVED** [review]
looks good, maybe increase multi thread count and decrease single thread loop?

**Comment by catenocrypt** [comment]
I adjusted as you suggested, though effective running times were not too high.  Now they are (on laptop):

```
▸     ✓ testSignOrderOneThread (0.171 seconds)
▸     ✓ testMultiThreadedSign (1.297 seconds)
▸     ✓ testDeriveOneThread (0.127 seconds)
▸     ✓ testMultiThreadedDerive (1.300 seconds)
▸ 	 Executed 349 tests, with 0 failures (0 unexpected) in 29.456 (29.648) seconds
```

**Pull request merged**